### ISSUE TITLE
Issue in file format conversion-Reformat function not working

### ISSUE THREAD

**Issue by Nagasundaram620** [jodyphelan/TBProfiler#187]
Hi 
   I used 
    tb-profiler reformat results/CMCVATB24.results.json --txt  
    but its showing the following error

# tb-profiler error report

* OS: linux
* Program version: 4.0.1
* Program call:
```
{'json': 'results/CMCVATB24.results.json', 'txt': True, 'csv': False, 'db': 'tbdb', 'external_db': None, 'func': <function main_reformat at 0x7f64287935e0>, 'prefix': 'CMCVATB24'}
```
## Traceback:
```
  File "/home/nagasundaram/miniconda3/bin/tb-profiler", line 594, in <module>
    args.func(args)
  File "/home/nagasundaram/miniconda3/bin/tb-profiler", line 308, in main_reformat
    tbp.write_text(results,conf,text_output)
  File "/home/nagasundaram/miniconda3/lib/python3.8/site-packages/tbprofiler/text.py", line 123, in write_text
    text_strings["missing_report"] = dict_list2text(json_results["qc"]["missing_positions"],["gene","locus_tag","position","position_type","drug_resistance_position"],sep=sep) if "gene_coverage" in json_results["qc"] else "Not available"
  File "/home/nagasundaram/miniconda3/lib/python3.8/site-packages/tbprofiler/text.py", line 35, in dict_list2text
    r = sep.join([variable2string(return_fields(row,x)) for x in headings])
  File "/home/nagasundaram/miniconda3/lib/python3.8/site-packages/tbprofiler/text.py", line 35, in <listcomp>
    r = sep.join([variable2string(return_fields(row,x)) for x in headings])
  File "/home/nagasundaram/miniconda3/lib/python3.8/site-packages/tbprofiler/text.py", line 22, in return_fields
    sub_obj = obj[largs[i]]
```
## Value:
```
'position_type'```

Kindly help me to solve this error

**Comment by jodyphelan** [comment]
Hi @Nagasundaram620 

I am not getting this error on my test files. Would you be able to email the json file so that I can recreate the error on my side?

**Comment by Nagasundaram620** [comment]
Dear Jodyphelan 

                        Thanks for the e-mail, here with i have attached the json file for your kind perusal. 

Thanks & Regards Dr. N. Nagasundaram Research Associate [ICMR]Vellore Institute of Technology (VIT)Vellore, TamilnaduIndia 
 

    On Monday, 15 November, 2021, 6:20:29 pm IST, Jody Phelan ***@***.***> wrote:  
 
 


Hi @Nagasundaram620

I am not getting this error on my test files. Would you be able to email the json file so that I can recreate the error on my side?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jodyphelan** [comment]
Hi @Nagasundaram620 I don't think the attachments are sent via github. Could you email it directly to jody.phelan@lshtm.ac.uk?

**Comment by jodyphelan** [comment]
There was a bug introduced in the latest version which caused the error. It has now been fixed and I will create a new version release soon.

**Comment by Nagasundaram620** [comment]
Hi jody
            Thanks for working on it. I tried to convert the json files to txt, The reformat function works for some files and not working for many json files. Its showing the same error. 

Here with i have attached the json file

# tb-profiler error report

* OS: linux
* Program version: 4.0.1
* Program call:
```
{'json': 'CMCVATB24.results1.json', 'txt': True, 'csv': False, 'db': 'tbdb', 'external_db': None, 'func': <function main_reformat at 0x7f03fec6a5e0>, 'prefix': 'CMCVATB24'}
```
## Traceback:
```
  File "/home/nagasundaram/miniconda3/bin/tb-profiler", line 594, in <module>
    args.func(args)
  File "/home/nagasundaram/miniconda3/bin/tb-profiler", line 308, in main_reformat
    tbp.write_text(results,conf,text_output)
  File "/home/nagasundaram/miniconda3/lib/python3.8/site-packages/tbprofiler/text.py", line 123, in write_text
    text_strings["missing_report"] = dict_list2text(json_results["qc"]["missing_positions"],["gene","locus_tag","position","position_type","drug_resistance_position"],sep=sep) if "gene_coverage" in json_results["qc"] else "Not available"
  File "/home/nagasundaram/miniconda3/lib/python3.8/site-packages/tbprofiler/text.py", line 35, in dict_list2text
    r = sep.join([variable2string(return_fields(row,x)) for x in headings])
  File "/home/nagasundaram/miniconda3/lib/python3.8/site-packages/tbprofiler/text.py", line 35, in <listcomp>
    r = sep.join([variable2string(return_fields(row,x)) for x in headings])
  File "/home/nagasundaram/miniconda3/lib/python3.8/site-packages/tbprofiler/text.py", line 22, in return_fields
    sub_obj = obj[largs[i]]
```
## Value:
```
'position_type'```

Thanks & Regards Dr. N. Nagasundaram Research Associate [ICMR]Vellore Institute of Technology (VIT)Vellore, TamilnaduIndia 
 

    On Thursday, 18 November, 2021, 2:09:14 pm IST, Jody Phelan ***@***.***> wrote:  
 
 


There was a bug introduced in the latest version which caused the error. It has now been fixed and I will create a new version release soon.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by pmenzel** [comment]
Just got the same problem here. :)

**Comment by jodyphelan** [comment]
Yes, The issue only appears when there is a sample with low coverage. The current release on conda doesn't have the fix yet. I'll work to release the fix today.

**Comment by Nagasundaram620** [comment]
Dear Jody 
               Thanks for your consistent support, I got the output for my sequences. 
Thanks & Regards Dr. N. Nagasundaram Research Associate [ICMR]Vellore Institute of Technology (VIT)Vellore, TamilnaduIndia 
 

    On Thursday, 18 November, 2021, 04:52:59 pm IST, Jody Phelan ***@***.***> wrote:  
 
 


Yes, The issue only appears when there is a sample with low coverage. The current release on conda doesn't have the fix yet. I'll work to release the fix today.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by jodyphelan** [comment]
No problem, I'm creating a new release for conda today

**Comment by jodyphelan** [comment]
This should be fixed in release [v4.0.3](https://github.com/jodyphelan/TBProfiler/releases/tag/v4.0.3) and is available on [bioconda](https://bioconda.github.io/recipes/tb-profiler/README.html).
